doc/source/development/extending.rst

```diff
@@ -135,6 +135,102 @@ by some other storage type, like Python lists.
+
+For ExtensionArrays backed by a single NumPy array, the
+:class:`~pandas.api.extensions.NDArrayBackedExtensionArray` class can save you
+some effort. It contains a private property ``_ndarray`` with the backing NumPy
+array and implements the extension array interface.
+
+Implement the following:
+
+``_box_func``
+  Convert from array values to the type you wish to expose to users.
+
+``_internal_fill_value``
+   Scalar used to denote ``NA`` value inside our ``self._ndarray``, e.g. ``-1``
+   for ``Categorical``, ``iNaT`` for ``Period``.
+
+``_validate_scalar``
+  Convert from an object to a value which can be stored in the NumPy array.
+
+``_validate_setitem_value``
+  Convert a value or values for use in setting a value or values in the backing
+  NumPy array.
+
+.. code-block:: python
+
+    class DateArray(NDArrayBackedExtensionArray):
+        _internal_fill_value = numpy.datetime64("NaT")
+
+        def __init__(self, values):
+            backing_array_dtype = "<M8[ns]"
+            super().__init__(values=values, dtype=backing_array_dtype)
+
+        def _box_func(self, value):
+            if pandas.isna(x):
+                return pandas.NaT
+            return x.astype("datetime64[us]").item().date()
+
+        def _validate_scalar(self, scalar):
+            if pandas.isna(scalar):
+                return numpy.datetime64("NaT")
+            elif isinstance(scalar, datetime.date):
+                return pandas.Timestamp(
+                    year=scalar.year, month=scalar.month, day=scalar.day
+                ).to_datetime64()
+            else:
+                raise TypeError("Invalid value type", scalar)
+
+        def _validate_setitem_value(self, value):
+            if pandas.api.types.is_list_like(value):
+                return np.array([self._validate_scalar(v) for v in value], dtype = self.dtype)
+            return self._validate_scalar(value)
+
+
+To support 2D arrays, use the ``_from_backing_data`` helper function when a
+method is called on multi-dimensional data of the same dtype as ``_ndarray``.
+
+.. code-block:: python
+
+    class CustomArray(NDArrayBackedExtensionArray):
+
+        ...
+
+        def min(self, *, axis: Optional[int] = None, skipna: bool = True, **kwargs):
+            pandas.compat.numpy.function.validate_minnumpy_validate_min((), kwargs)
+            result = pandas.core.nanops.nanmin(
+                values=self._ndarray, axis=axis, mask=self.isna(), skipna=skipna
+            )
+            if axis is None or self.ndim == 1:
+                return self._box_func(result)
+            return self._from_backing_data(result)
+
+
+Subclass the tests in :mod:`pandas.tests.extension.base` in your test suite to
+validate your implementation.
+
+.. code-block:: python
+
+    @pytest.fixture
+    def data():
+        return CustomArray(numpy.arange(-10, 10, 1)
+
+
+    class Test2DCompat(base.NDArrayBacked2DTests):
+        pass
+
+
+    class TestComparisonOps(base.BaseComparisonOpsTests):
```

Comment:
we've moved to using `base.ExtensionTests` instead of a bunch of classes

Reply:
changed to using that